### PR TITLE
 Upgrade login API version

### PR DESCRIPTION
**Pull request by AdriB02**
# Upgrade the version of the login API to fix login issues. To be tested

## Description

The main goal of this pull request is to update the login api because of login issues. Other UI improvement have been added, such as new LoadingPage. This pull request includes several updates to the project, focusing on dependency management, testing enhancements, resource additions, and UI adjustments.

### What has been changed?

> ### Authentication chages :
>  The main goal of this PR is to update the SignInScreen to leverage the new Google sign-in process with CredentialManager and refines the user interface with animated effects. The updated screen introduces the following key changes:
>  * Replaced the Firebase Google sign-in process with CredentialManager, supporting a more secure, simplified authentication flow.
>   * Uses GetGoogleIdOption and GetCredentialRequest objects for streamlined Google credential requests.
>   * New functions, googleSignInRequest and handleSignIn, manage credential requests and authentication with Firebase.
>   * Enhance error handling using GetCredentialException.
> For more information, you can refer to the explanation provided by Google here: [Credential Manager Overview on Android Developers](https://developer.android.com/identity/sign-in/credential-manager-siwg).

> ### UI Adjustments:
> Rename the `SignIn.kt` file to `SignInScreen.kt`, incorporating the updated Google Sign-In UI and logic. Upgrade the login API to the latest version of Google Login, and align the UI design to match the Figma specifications. 
> * Add a new animated screen background using various animated food icons.
> * Update the main screen layout with a new animated "cook" image and revamped button styles.
> * Add animation functions (RecipesAnimation and AnimatedImage) to create a lively background, enriching the user experience.
> A demo of the updated login screen is provided here:

> https://github.com/user-attachments/assets/de2995b6-4aa6-4312-838e-ba88f222e283

> ### Testing Enhancements:
> * Updated `SignInTest.kt` to include coroutine testing utilities and modified test cases to use `useUnmergedTree` for node assertions. [[1]](diffhunk://#diff-8bf35c40d396f626356384159ae1b7991c1bf96297f1946702d52764901c70edL11-R18) [[2]](diffhunk://#diff-8bf35c40d396f626356384159ae1b7991c1bf96297f1946702d52764901c70edL48-R89)
> * Added new test cases in `LoadingAnimationTest.kt` to verify fruit creation, properties, and completion triggers.

> ### Others Improvements
> * Updated text style in `Type.kt` to make `titleLarge` bold and increase its size to better align with the SwipePage title of Figma. This change was possible because the style was only used in this context.
> * Added `googleid` to the dependencies in `app/build.gradle.kts`.

> ### SonarCloud issue
> `Dispatchers.Main` is a dispatcher that confines coroutines to the main thread. This ensures that any changes to the sign-in process (such as updating the UI to reflect success or failure) are performed safely and efficiently.

### Why was this change made?

> This change resolves issues with the previous Google login system, where login failures occurred without clear explanations. By integrating CredentialManager with the updated Google sign-in flow, the solution now offers a more reliable and secure user experience. Additionally, the new animated interface adds visual appeal, making the sign-in process both smoother and more engaging for users.

## Checklist
- [x] Tests have been added to ensure loggin.
- [x] Documentation has been updated to reflect the latest changes.
- [x] All CI checks passed.
- [x] Linked issue/feature (if applicable): #154

### CODE REVIEW

File: app/src/main/res/drawable/tomato.png
Comment (RagdollGoDown) [inline app/src/main/res/drawable/tomato.png:1]: All of these images are way bigger than they need to be. You should compress them, they should be at most 200kb as these are small icons. Unless you wanted to use them for something where we would need this quality.

**Review by RagdollGoDown — CHANGES_REQUESTED** [review]
I absolutely love the new entry animation! It really enhance our user experience. Now for the bad, I do need some changes to be made. 
First, the images are well picked, but aren't compressed and take up a lot more space than they should.
Second, there is an issue mentioned in the sonar cloud analysis that mentions hardcoded dispatchers. I think it would be a good idea to apply the correction given on sonar cloud.
Third, don't do this now, but it would have been better to do two PRs; one for the new signin methods and another for the UI animations.
Otherwise, the rest of the code is readable and nicely organized. Once these changes will have been made, then I think that it's ready to be merged.

**Comment by AdriB02** [comment]
Thank you for the feedback! I'm really glad you like the animation and that it enhances the user experience. I appreciate your suggestions for improvements, and here are my responses:

Image Compression: I understand your concern about the image sizes. I opted for larger images to ensure they look good on bigger screens, like iPads. However, I will work on compressing them to bring the file sizes under 200kb while maintaining as much quality as possible.

SonarCloud Issue with Dispatchers: I noticed the note about the hardcoded dispatchers. I’ve added a comment explaining that Dispatchers.Main ensures UI updates are handled safely on the main thread. Could you please take a look at that comment and let me know if it can be accepted?

Separate PRs for Features: I combined the sign-in methods and UI animations in this PR because we previously discussed submitting larger PRs in our last meeting. Since the animation is part of the sign-in experience, I thought it made sense to include both. However, I understand that keeping them separate could make the review process easier, and I’ll consider that for future PRs.

Thanks again for your valuable feedback!

**Review by RagdollGoDown — APPROVED** [review]
Now that the changes have been made, we can merge the branch.

**Pull request merged**